### ISSUE TITLE
get_relations_by_prefix only works in target schema

### ISSUE THREAD

**Issue by clausherther** [fishtown-analytics/dbt-utils#277]
FYI, we are currently using `{% set relations = dbt_utils.get_relations_by_prefix('xa', '') %}` to find all tables in the `xa` schema, but this returns 0 rows now if the model you're running this in (`{{adapter.quote(database)}}.{{schema}}`) is in a different schema than `schema_pattern`. I _think_ this is a bug?

```
select distinct
            table_schema, table_name

        from {{adapter.quote(database)}}.{{schema}}.INFORMATION_SCHEMA.TABLES
        where table_schema = '{{schema_pattern}}'
            and lower(table_name) like lower ('{{table_pattern}}')
            and lower(table_name) not like lower ('{{exclude}}')
```

_Originally posted by @clausherther in https://github.com/fishtown-analytics/dbt-utils/pull/275#issuecomment-689251250_

**Comment by clausherther** [comment]
For example, if I use 
```
{% set relations = dbt_utils.get_relations_by_prefix('xa', '') %}
```
in a model materialized in `audit` b/c I'm running the model during an audit run, the above macro will return 0 rows.

**Comment by companywatch-rob** [comment]
I am seeing this issue too.  I use dbt_utils.get_relations_by_prefix to look for tables from another project/dataset by passing in these values as parameters.  These seem to be being ignored.

I think the issue is in the macro get_tables_by_pattern_sql.  The call to get_tables_by_pattern_sql passes target.database as the value for database rather then the value passed in.

```
{% macro get_tables_by_pattern_sql(schema_pattern, table_pattern, exclude='', database=target.database) %}
    {{ adapter.dispatch('get_tables_by_pattern_sql', packages = dbt_utils._get_utils_namespaces())
        (schema_pattern, table_pattern, exclude='', database=target.database) }} 
{% endmacro %}
```

Also, in the bigquery macro implementation, it uses the value of {{schema}} in the from clause rather than {{schema_pattern}}.  I don't know where the value for {{schema}} comes from - default? 

```
{% macro bigquery__get_tables_by_pattern_sql(schema_pattern, table_pattern, exclude='', database=target.database) %}
    
        select distinct
            table_schema, table_name

        from {{adapter.quote(database)}}.{{schema}}.INFORMATION_SCHEMA.TABLES
        where table_schema = '{{schema_pattern}}'
            and lower(table_name) like lower ('{{table_pattern}}')
            and lower(table_name) not like lower ('{{exclude}}')

{% endmacro %}
```

**Comment by jtcohen6** [comment]
Thanks for those details @clausherther @companywatch-rob. This is indeed a bug/regression introduced in 0.6.0, as a result of refactoring + deprecation work we did around the `get_x_by_y` complex. I believe the resolution here is https://github.com/fishtown-analytics/dbt-utils/pull/275, which we'll merge and release shortly as 0.6.1.

**Comment by clrcrl** [comment]
Closed by #275
